container/podman/handler.go

```diff
@@ -0,0 +1,308 @@
+	// If the NetworkMode starts with 'container:' then we need to use the IP address of the container specified.
+	// This happens in cases such as kubernetes where the containers doesn't have an IP address itself and we need to use the pod's address
+	networkMode := string(handler.networkMode)
+	if handler.ipAddress == "" && strings.HasPrefix(networkMode, "container:") {
+		id := strings.TrimPrefix(networkMode, "container:")
+		ctnr, err := InspectContainer(id)
+		if err != nil {
+			return nil, err
+		}
+		handler.ipAddress = ctnr.NetworkSettings.IPAddress
+	}
+
+	if metrics.Has(container.DiskUsageMetrics) {
+		handler.fsHandler = &docker.FsHandler{
+			FsHandler:       common.NewFsHandler(common.DefaultPeriod, rootfsStorageDir, otherStorageDir, fsInfo),
+			ThinPoolWatcher: thinPoolWatcher,
+			ZfsWatcher:      zfsWatcher,
+			DeviceID:        ctnr.GraphDriver.Data["DeviceId"],
+			ZfsFilesystem:   zfsFilesystem,
+		}
+	}
+
+	// Split env vars to get metadata map.
+	for _, exposedEnv := range metadataEnvAllowList {
+		if exposedEnv == "" {
+			continue
+		}
+
+		for _, envVar := range ctnr.Config.Env {
+			if envVar != "" {
+				splits := strings.SplitN(envVar, "=", 2)
+				if len(splits) == 2 && strings.HasPrefix(splits[0], exposedEnv) {
+					handler.envs[strings.ToLower(splits[0])] = splits[1]
+				}
+			}
+		}
+	}
+
+	return handler, nil
+}
+
+func (p podmanContainerHandler) ContainerReference() (info.ContainerReference, error) {
+	return p.reference, nil
+}
+
+func (p podmanContainerHandler) needNet() bool {
+	if p.metrics.Has(container.NetworkUsageMetrics) {
+		p.networkMode.IsContainer()
+		return !p.networkMode.IsContainer()
+	}
+	return false
+}
+
+func (p podmanContainerHandler) GetSpec() (info.ContainerSpec, error) {
+	hasFilesystem := p.metrics.Has(container.DiskUsageMetrics)
+
+	spec, err := common.GetSpec(p.cgroupPaths, p.machineInfoFactory, p.needNet(), hasFilesystem)
+	if err != nil {
+		return info.ContainerSpec{}, err
+	}
+
+	spec.Labels = p.labels
+	spec.Envs = p.envs
+	spec.Image = p.image
+	spec.CreationTime = p.creationTime
+
+	return spec, nil
+}
+
+func (p podmanContainerHandler) GetStats() (*info.ContainerStats, error) {
+	stats, err := p.libcontainerHandler.GetStats()
+	if err != nil {
+		return stats, err
+	}
+
+	if !p.needNet() {
+		stats.Network = info.NetworkStats{}
+	}
+
+	err = docker.GetFsStats(stats, p.machineInfoFactory, p.metrics, p.storageDriver,
+		p.fsHandler, p.fsInfo, p.thinPoolName, p.rootfsStorageDir, p.zfsParent)
+	if err != nil {
+		return stats, err
+	}
+
+	return stats, nil
+}
+
+func (p podmanContainerHandler) ListContainers(listType container.ListType) ([]info.ContainerReference, error) {
+	return []info.ContainerReference{}, nil
+}
+
+func (p podmanContainerHandler) ListProcesses(listType container.ListType) ([]int, error) {
+	return p.libcontainerHandler.GetProcesses()
+}
+
+func (p podmanContainerHandler) GetCgroupPath(resource string) (string, error) {
```

Comment:
Why not just CgroupPath?

Reply:
`podmanContainerHandler` implements `ContainerHandler` interface. I agree that it should be refactored but maybe in a different PR because the current one is huge.